opm/input/eclipse/Schedule/Well/Well.cpp

```diff
@@ -363,7 +363,7 @@ Well::Well(const RestartIO::RstWell& rst_well,
         p->addProductionControl(Well::ProducerCMode::BHP);
         if (! p->predictionMode) {
             p->BHPTarget.update(0.0);
-            p->setBHPLimit(rst_well.bhp_target_double);
+            p->setBHPLimit(rst_well.bhp_target_float);
```

Comment:
I seem to recall that `WellProductionProperties::setBHPLimit()` expects an [SI unit](https://github.com/OPM/opm-common/blob/fc6aca0bec0d861a1f86e36e0db7b7e15fd52dd0/opm/input/eclipse/Schedule/Well/WellProductionProperties.cpp#L392-L394) input value.  Our handling of [`FBHPDEF`](https://github.com/OPM/opm-common/blob/fc6aca0bec0d861a1f86e36e0db7b7e15fd52dd0/opm/input/eclipse/Schedule/KeywordHandlers.cpp#L127-L128) backs that up because the internalised/SI values are [passed directly into](https://github.com/OPM/opm-common/blob/fc6aca0bec0d861a1f86e36e0db7b7e15fd52dd0/opm/input/eclipse/Schedule/Well/WellKeywordHandlers.cpp#L116-L126) `WellProductionProperties::handleWCONHIST()` which in turn [calls `setBHPLimit()`](https://github.com/OPM/opm-common/blob/fc6aca0bec0d861a1f86e36e0db7b7e15fd52dd0/opm/input/eclipse/Schedule/Well/WellProductionProperties.cpp#L249) with those values unless the `WCONHIST` record defines a separate BHP limit.

When restarting a simulation run, `bhp_target_float` is stored in output units while `bhp_target_double` is stored in SI units.  I will freely admit that that is **very** confusing and there might very well be programming errors lurking here, but I do think we should continue to pass `bhp_target_double` in just this location.  If you have evidence to suggest otherwise then that may point to a deeper problem.

Reply:
I have a field case where the double value is clearly wrong, so pretty sure we need the float value here (but with the units, which I just didn't think of - will fix and check)

Reply:
> have a field case where the double value is clearly wrong, so pretty sure we need the float value here (but with the units, which I just didn't think of - will fix and check)

Oh, I **think** I see a potential issue in our current implementation.  We load `bhp_target_double` [from `XWEL`](https://github.com/OPM/opm-common/blob/fc6aca0bec0d861a1f86e36e0db7b7e15fd52dd0/opm/io/eclipse/rst/well.cpp#L176)&ndash;i.e., from the solution variables&ndash;instead of `SWEL` (the targets/limits defined in the `WCON*` keyword).  On the other hand `bhp_target_float` is [loaded from `SWEL`](https://github.com/OPM/opm-common/blob/fc6aca0bec0d861a1f86e36e0db7b7e15fd52dd0/opm/io/eclipse/rst/well.cpp#L130) which would normally be a better data source.  At the minute I can't remember why we load `_double` from `XWEL` instead of `SWEL` so I'll have to go back in the development history to get a reminder.

Reply:
Ok, thanks. With correct units things now look OK on the case I have at hand.